### DISCUSSION TITLE
Unable to find git directory on install

### DISCUSSION THREAD

**Discussion by forrestab** [alirezanet/Husky.Net#44] [Q&A] [answered]
I wasnt sure if this should be an issue, but I saw the discussions tab so I figured I would start here.

Here is my current structure is:

```
|-- .git
|-- src
|   |-- *.sln
|   |-- Api
|   |   |-- .config
| ...
```

Ive installed the husky tool to the Api project and when I run `dotnet husky install` it throws the following error:

> Failed to start a process with file path 'git'. Target file or working directory doesn't exist or the provided credentials are invalid.

I assume this is because there is no `.git` directory in the current working directory. I didnt see it in the documentation (I could be missing it), but can we provide a custom path to the `.git` directory during install?

**Comment by alirezanet** [comment, marked as answer]
Hi @forrestab,
The error is not related to the `.git` folder, I think the problem is related to the git installation on your machine, try to reinstall the git and make sure that it's accessible from your terminal. If you're using windows add the git installation path to the System Environment.

**Reply by alirezanet** [reply]
> Ive installed the husky tool to the Api project 

I also should note that you must install husky into your root project usually, husky is not a dependency it is an extension for your git repo.

**Reply by forrestab** [reply]
I dont know why I didnt think about the git executable not being in PATH. Ive added it and am now able to use `git` from the terminal. However, Im now seeing the below error which I believe is probably in regards to your second comment, correct?

> .git can't be found (see https://alirezanet.github.io/Husky.Net/guide/getting-started)
Git hooks installation failed

I think I may have gotten confused with the phrase "root project." I saw the word "project" and thought it had to exist in the same directory as a project, but I think what you are saying is the tool needs to be installed in the repository root, next to the .git folder or do I still have that wrong?

**Reply by forrestab** [reply]
That ended up being the issue, after moving the tool install to the root of the repository it installed successfully. Thanks for your help.
